src/solver/simulation/solver.hxx

```diff
@@ -344,7 +345,10 @@ void ISimulation<Impl>::run()
         }
 
         if (parameters.useCustomScenario)
+        {
             ApplyCustomScenario(study);
+            FinalReservoirLevelPreChecks(study);
+        }
```

Comment:
Include `FinalReservoirLevelPreChecks` (the 3 replacement functions) into `ApplyCustomScenario`